src/components/Dropdown/DropdownWithIcon.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { settings } from '../../constants/Settings';
+
+import { Dropdown } from './index';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  itemToString: PropTypes.func,
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      text: PropTypes.string.isRequired,
+      icon: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
+    })
+  ).isRequired,
+};
+
+const defaultPropTypes = {
+  itemToString: null,
+};
+
+const DropdownWithIcon = ({
+  selectedViewId,
```

Comment:
You are not using this one. It should probably be renamed and then passed down to the Dropdown as the prop `selectedItem`